syncs/shardvalue_test.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package syncs
+
+import (
+	"math"
+	"runtime"
+	"sync"
+	"sync/atomic"
+	"testing"
+
+	"golang.org/x/sys/cpu"
+)
+
+func TestShardValue(t *testing.T) {
+	type intVal struct {
+		atomic.Int64
+		_ cpu.CacheLinePad
+	}
```

Comment:
Why define a new type and not use `intShard`?

Reply:
because i don't want to couple the two parts so tightly, they could be split into packages later, this keeps the dependency linear from shardint -> shardval, and not back again